pkg/svcat/service-catalog/broker.go

```diff
@@ -120,11 +129,57 @@ func (sdk *SDK) Register(brokerName string, url string, opts *RegisterOptions) (
 		}
 
 	}
-	request := &v1beta1.ClusterServiceBroker{
+	if scopeOpts.Scope.Matches(ClusterScope) {
+		request := &v1beta1.ClusterServiceBroker{
+			ObjectMeta: v1.ObjectMeta{
+				Name: brokerName,
+			},
+			Spec: v1beta1.ClusterServiceBrokerSpec{
+				CommonServiceBrokerSpec: v1beta1.CommonServiceBrokerSpec{
+					CABundle:              caBytes,
+					InsecureSkipTLSVerify: opts.SkipTLS,
+					RelistBehavior:        opts.RelistBehavior,
+					RelistDuration:        opts.RelistDuration,
+					URL:                   url,
+					CatalogRestrictions: &v1beta1.CatalogRestrictions{
+						ServiceClass: opts.ClassRestrictions,
+						ServicePlan:  opts.PlanRestrictions,
+					},
+				},
+			},
+		}
+		if opts.BasicSecret != "" {
+			request.Spec.AuthInfo = &v1beta1.ClusterServiceBrokerAuthInfo{
+				Basic: &v1beta1.ClusterBasicAuthConfig{
+					SecretRef: &v1beta1.ObjectReference{
+						Name:      opts.BasicSecret,
+						Namespace: opts.Namespace,
+					},
+				},
+			}
+		} else if opts.BearerSecret != "" {
+			request.Spec.AuthInfo = &v1beta1.ClusterServiceBrokerAuthInfo{
+				Bearer: &v1beta1.ClusterBearerTokenAuthConfig{
+					SecretRef: &v1beta1.ObjectReference{
+						Name:      opts.BearerSecret,
+						Namespace: opts.Namespace,
+					},
+				},
+			}
+		}
+
+		result, err := sdk.ServiceCatalog().ClusterServiceBrokers().Create(request)
+		if err != nil {
+			return nil, fmt.Errorf("register request failed (%s)", err)
+		}
+
+		return result, nil
+	}
+	request := &v1beta1.ServiceBroker{
```

Comment:
I recommend wrapping this in an if statement, just to ensure that it's only executed for scope = namespace, and to make it clear to others when it's used.